common/djangoapps/student/views.py

```diff
@@ -712,6 +714,58 @@ def dashboard(request):
     return render_to_response('dashboard.html', context)
 
 
+@login_required
+@ensure_csrf_cookie
+def revisions(request):
+    """
+    Return a JSON list of all revisions for a user. Each revision includes a name, due date, and URL.
+    """
+    user = request.user
+
+    revisions = _get_pending_revisions(user)
+    json_revisions = json.dumps(revisions)
+
+    return HttpResponse(json_revisions)
+
+
+def _get_pending_revisions(user):
+    """
+    Returns information about each problem that needs revision for a given user, including a display name, the due date
+    of the revision, and a courseware URL.
+    """
+    # TODO: Come up with a list of locators by grabbing all of the AdaptiveLibraryContentModules and making a request
+    #       to the Domoscio API.
+    serialized_locators = [
+        'block-v1:asdf+asdf+asdf+type@problem+block@9adc878ed55943caa7d4b4cd48e52752',
+        'block-v1:asdf+asdf+asdf+type@problem+block@0c4c0798cf3847ae9c706cdcb9442c62',
+    ]
+
+    locators = map(BlockUsageLocator.from_string, serialized_locators)
+
+    revision_data = []
+    for locator in locators:
+        # Resolve the courseware URL for the revision xblock.
+        (
+            course_key, chapter, section, vertical_unused,
+            position, final_target_id_unused
+        ) = path_to_location(modulestore(), locator)
+
+        url = reverse(
+            'courseware_position',
+            args=(unicode(course_key), chapter, section, navigation_index(position))
+        )
+
+        xblock = modulestore().get_item(locator)
+        # TODO: Use information from the fields in the xblock to populate the name and date below.
```

Comment:
@bdero The TODOs in this function still don't describe the Python API approach that we want to use. I'm fine with leaving them as is but we'll need to make sure that whoever takes OC-2154 knows this :)